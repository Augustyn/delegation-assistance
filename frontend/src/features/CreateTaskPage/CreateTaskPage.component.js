import React from "react";
import { reduxForm, Form } from "redux-form";
import { func } from "prop-types";

import LayoutMain from "../../components/layouts/LayoutMain/LayoutMain.container";
import Input from "../../components/Input/Input.component";
import Button from "../../components/Button/Button.component";
import Textarea from "../../components/Textarea/Textarea.component";
import Card from "../../components/Card/Card.component";

export const CreateTaskPage = props => {
  const { handleSubmit } = props;
  return (
    <LayoutMain title="Checklist management">
      <div className="container">
        <Card title="Add new task">
          <Form onSubmit={handleSubmit}>
            <Input name="name" label="Task name" />
            <Textarea name="description" type="textarea" label="Task description" />
            <Button text="Add task" type="submit" />
          </Form>
        </Card>
      </div>
    </LayoutMain>
  );
};

CreateTaskPage.propTypes = {
  handleSubmit: func
};

export default reduxForm({
  form: "create-task"
})(CreateTaskPage);
