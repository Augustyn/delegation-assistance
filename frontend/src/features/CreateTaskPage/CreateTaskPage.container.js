import React, { Component } from "react";
import { connect } from "react-redux";
import { func } from "prop-types";

import CreateTaskPage from "./CreateTaskPage.component";
import { addTask } from "../../actions/checklists.actions";

class CreateTaskPageContainer extends Component {
  static propTypes = {
    addTask: func
  };

  _handleSubmit = values => {
    this.props.addTask(values.name, values.description).then(() => (window.location.href = "/checklist"));
  };

  render() {
    return <CreateTaskPage onSubmit={this._handleSubmit} />;
  }
}

const mapDispatchToProps = {
  addTask
};

export default connect(
  null,
  mapDispatchToProps
)(CreateTaskPageContainer);
