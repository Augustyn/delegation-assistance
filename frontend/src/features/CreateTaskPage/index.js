import CreateTaskPage from "./CreateTaskPage.container";
export default CreateTaskPage;
