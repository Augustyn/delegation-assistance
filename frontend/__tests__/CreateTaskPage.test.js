import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import { CreateTaskPage } from "../src/features/CreateTaskPage/CreateTaskPage.component";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

Enzyme.configure({ adapter: new Adapter() });

const setup = () => {
  const props = {
    handleSubmit: jest.fn(),
    subbmiting: false,
    invalid: false
  };

  const enzymeWrapper = shallow(<CreateTaskPage {...props} />);

  return {
    props,
    enzymeWrapper
  };
};

describe("Checklists Page", () => {
  let enzymeWrapper;
  let props;

  beforeEach(() => {
    ({ enzymeWrapper, props } = setup());
  });

  it("should render correctly", () => {
    expect(enzymeWrapper).toMatchSnapshot();
  });

  it("should call submit with task object", () => {
    const value = {
      name: "Task name",
      desciption: "Task description"
    };
    enzymeWrapper.find("Form").simulate("submit", value);
    expect(props.handleSubmit).toBeCalledWith(value);
  });
});
