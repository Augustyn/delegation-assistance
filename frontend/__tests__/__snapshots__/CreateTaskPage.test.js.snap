// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checklists Page should render correctly 1`] = `
<Connect(LayoutMainContainer)
  title="Checklist management"
>
  <div
    className="container"
  >
    <Card
      title="Add new task"
    >
      <Form
        onSubmit={[MockFunction]}
      >
        <Input
          label="Task name"
          name="name"
        />
        <Textarea
          label="Task description"
          name="description"
          type="textarea"
        />
        <Button
          text="Add task"
          type="submit"
        />
      </Form>
    </Card>
  </div>
</Connect(LayoutMainContainer)>
`;
